Source/JavaScriptCore/runtime/JSGlobalObject.cpp

```diff
@@ -1162,7 +1162,7 @@ void JSGlobalObject::init(VM& vm)
     ShadowRealmConstructor* shadowRealmConstructor = ShadowRealmConstructor::create(vm, ShadowRealmConstructor::createStructure(vm, this, m_functionPrototype.get()), m_shadowRealmPrototype.get());
     m_shadowRealmConstructor.set(vm, this, shadowRealmConstructor);
 
-    RegExpConstructor* regExpConstructor = RegExpConstructor::create(vm, RegExpConstructor::createStructure(vm, this, m_functionPrototype.get()), m_regExpPrototype.get());
+    RegExpConstructor* regExpConstructor = RegExpConstructor::create(vm, this, RegExpConstructor::createStructure(vm, this, m_functionPrototype.get()), m_regExpPrototype.get());
```

Comment:
yeah i realized i could do it a different way after making this change

im not sure which is better/preferred, as ive seen it be done both ways (and have personally done it this way at least once).  im pretty sure the `JSGlobalObject` used to create the `RegExpPrototype` is the same, but i figured it couldn't hurt to pass it along explicitly just in case :)

let me know if it's better to not do this